src/ast/passes/config_analyser.cpp

```diff
@@ -165,26 +166,33 @@ void ConfigAnalyser::visit(StackMode &mode)
 void ConfigAnalyser::visit(AssignConfigVarStatement &assignment)
 {
   Visitor<ConfigAnalyser>::visit(assignment);
-  std::string &raw_ident = assignment.config_var;
+  std::string &raw_ident = assignment.config_var->ident;
 
   std::string err_msg;
   const auto maybeConfigKey = bpftrace_.config_->get_config_key(raw_ident,
                                                                 err_msg);
 
   if (!maybeConfigKey.has_value()) {
-    assignment.addError() << err_msg;
+    assignment.config_var->addError() << err_msg;
     return;
   }
 
   if (!assignment.expr->is_literal) {
-    assignment.addError() << "Assignment for " << assignment.config_var
-                          << " must be literal.";
+    assignment.expr->addError()
+        << "Assignment for " << assignment.config_var->ident
+        << " must be literal.";
     return;
   }
 
   auto configKey = maybeConfigKey.value();
 
   std::visit([&](auto key) { set_config(assignment, key); }, configKey);
+
+  if (bpftrace_.config_->is_unstable(configKey)) {
+    assignment.config_var->addWarning()
+        << "Script is using an unstable feature: "
+        << raw_ident.erase(0, UNSTABLE_PREFIX.size());
```

Comment:
Why is the logic for `is_unstable` split across translation units like this? I would either make the analyzer aware of these things, or limit it to the config file itself. The export of `UNSTABLE_PREFIX` is the part that I'm complaining about here, since it seems firmly in the "implementation detail" realm, with `is_unstable` being a public API.

Reply:
Yeah maybe I'll just remove the prefix erasing here.